### DISCUSSION TITLE
Profiled app crashes randomly only when profiled using API (AsyncProfiler.java)

### DISCUSSION THREAD

**Discussion by tkowalcz** [async-profiler/async-profiler#662] [Q&A]
Hi,

async-profiler is an awesome tool and I was working to build it into one of our internal applications so it can profile itself. When I profile it using profile.sh (e.g.: ` -d 10 -e alloc -o flamegraph  -f ~/flamegraph.svg PID`) everything works fine. But when I use AsyncProfiler.java from within the application it will crash almost instantly in a spectacular fashion (e.g. SIGSEGV in JVM code).

This happens on M1 mac with various recent async profiler versions. It also seem to crush when run on google cloud. 

I understand that this is not too much information and I do not expect anyone to fix it for me. Maybe there is some significant difference between how profiler works when using that API? Or there is something that I should do first? Should I RTFM that I missed?

Thanks for any valid pointers ;)

**Comment by apangin** [comment]
Can you please share a crash report `hs_err.log`? Would be even better if you attach several examples.

**Comment by tkowalcz** [comment]
Right now when I run the app on command line, after triggering profiling, I get this in terminal:

`[1]    602 killed XXX` where XXX is the command line of my Java app. Never seen something like this.

I will try to generate some legit JVM crashes and gether `hs_err` files.

**Reply by apangin** [reply]
BTW, what is JDK version?
```
java -version
java -Xinternalversion
```
